databroker/core.py

```diff
@@ -476,7 +623,7 @@ def to_dask(self):
     def _close(self):
         self.bag = None
 
-    def canonical(self, *, fill):
+    def canonical(self, *, fill, strict_order=False):
```

Comment:
Did we discuss the best default value for this? My immediate gut feeling is that it should be `True`. Consumers expects documents in the order they were received from the `RunEngine` may be sensitive to this. Advanced users can choose `False` if they are sure it is safe to do so in order to get faster throughput.

Reply:
updated